app/controllers/comments_controller.rb

```diff
@@ -1,48 +1,61 @@
 class CommentsController < ApplicationController
   before_action :authenticate_user!
-  before_action :set_comment, only: [:show, :edit, :update, :destroy]
+  before_action :set_sensitivites, only: [:create]
+  after_action :verify_authorized
 
   respond_to :html
 
-  def index
-    @comments = Comment.all
-    respond_with(@comments)
-  end
+  # def index
+  #   @comments = Comment.all
+  #   respond_with(@comments)
+  # end
 
-  def show
-    respond_with(@comment)
-  end
+  # def show
+  #   respond_with(@comment)
+  # end
 
-  def new
-    @comment = Comment.new
-    respond_with(@comment)
-  end
+  # def new
+  #   @comment = Comment.new
+  #   respond_with(@comment)
+  # end
 
-  def edit
-  end
+  # def edit
+  # end
 
   def create
-    @comment = Comment.new(comment_params)
+    authorize Comment
+    @comment = Comment.new(augmented_comment_params)
     @comment.save
     respond_with(@comment)
   end
 
-  def update
-    @comment.update(comment_params)
-    respond_with(@comment)
-  end
+  # def update
+  #   @comment.update(comment_params)
+  #   respond_with(@comment)
+  # end
 
-  def destroy
-    @comment.destroy
-    respond_with(@comment)
-  end
+  # def destroy
+  #   @comment.destroy
+  #   respond_with(@comment)
+  # end
 
   private
-    def set_comment
-      @comment = Comment.find(params[:id])
+    # def set_comment
+    #   @comment = Comment.find(params[:id])
+    # end
+
+    def augmented_comment_params
+        # Ensure that the event is created with correct user and correct organization (no mistification)
+        # We must also insert the event_id, but I'm not sure how
+        new_params = comment_params.merge(owner_organization: current_user.organization.id, user: current_user)
+    end
+
+    def set_sensitivites
+      sensitivitiy_hash =  Comment.sensitivities
+      @sensitivities = sensitivitiy_hash.select { |s| sensitivitiy_hash[s] <= sensitivitiy_hash[current_user.clearance] }
     end
 
     def comment_params
-      params.require(:comment).permit(:user_id, :body, :event_id, :sensitivity, :level)
+      params.require(:comment).permit(:body, :sensitivity, :level)
```

Comment:
Mas isso não permitirá que o utilizador o altere, criando all kinds of
shit? (Embora assuma que na form o possas colocar read only)
(tinha pensado em nem sequer o pedir, e enfiá-lo automaticamente no
controller, mas levantava-se o problema de como fazer a variável chegar ao
controlador.

On Sat, Jan 10, 2015, 10:23 PM Diogo Mónica <notifications@github.com>
wrote:

> In app/controllers/comments_controller.rb
> <https://github.com/pmonica/elog/pull/7#discussion-diff-22762449>:
>
> >      end
> >
> >      def comment_params
> > -      params.require(:comment).permit(:user_id, :body, :event_id, :sensitivity, :level)
> > +      params.require(:comment).permit(:body, :sensitivity, :level)
>
> Event has to come as Event ID
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pmonica/elog/pull/7/files#r22762449>.
>